core/ledger/validator.go

```diff
@@ -184,3 +185,13 @@ func HeaderCheck(header *Header, receiveTime int64) error {
 
 	return nil
 }
+
+func TimestampCheck(timestamp int64) error {
```

Comment:
How about single line
```
abs(time.Now().Unix() - timestamp) <= TimestampTolerance
```
for more clearer?

Reply:
There are 2 reasons I tried to avoid this:
1. I plan to make the timestamp check asymmetric because block propagation will only add delay, so we should allow more tolerance before.
2. In general I try to avoid doing computation on any user provide data to prevent potential bug like overflow (this one seems to be fine though).